src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -899,6 +974,52 @@ private LfNetwork create(int numCC, int numSC, Network network, List<Bus> buses,
         return lfNetwork;
     }
 
+    /**
+     * If we perform Area Interchange Control, we need all buses to have an Area or to be boundary buses
+     * A boundary bus is connected to one or multiple buses that are all in areas different from each other
+     * If a bus is not in any area, and is not a boundary bus, the area interchange control could not be performed and an exception will be thrown
+     * However, it is possible that the flow through this bus is not considered for the interchange flow of some of the areas it is connected to.
+     * In this case, the slack injection of this bus will be shared between those areas to ensure a correct balancing
+     */
+    // TODO add reports
+    private static void checkBusesWithNoArea(LfNetwork lfNetwork, LfNetworkParameters parameters, LoadingContext loadingContext) {
+        if (parameters.isAreaInterchangeControl()) {
+            List<LfBus> handledBuses = lfNetwork.getAreas().stream().flatMap(area ->
+                    area.getBuses().stream())
+                    .toList();
+            List<LfBus> unhandledBuses = lfNetwork.getBuses().stream().filter(b -> !handledBuses.contains(b)).toList();
```

Comment:
performance issue here, especially the `.filter(b -> !handledBuses.contains(b))` because handledBuses is a List.

changing to a Set could be an option, but we can see here that we need in LfBus a method like

`Optional<LfArea> getLfArea()`

Reply:
adressed in 9a9478c80af6e5e216bc90e464be48e7287c92c6